src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -154,41 +156,82 @@ private void ReadData()
         /// </returns>
         private bool ReadThroughOneWire()
         {
+            byte readVal = 0;
+            uint count;
+
+            // keep data line HIGH
+            _controller.SetPinMode(_pin, PinMode.Output);
+            _controller.Write(_pin, PinValue.High);
+            DelayHelper.DelayMilliseconds(20, true);
+
+            // send trigger signal
             _controller.Write(_pin, PinValue.Low);
-            // wait 18 milliseconds
-            DelayHelper.DelayMilliseconds(18, true);
+            // wait at least 18 milliseconds
+            // here wait for 18 milliseconds will cause sensor initialization to fail
+            DelayHelper.DelayMilliseconds(20, true);
 
+            // pull up data line
             _controller.Write(_pin, PinValue.High);
             // wait 20 - 40 microseconds
-            DelayHelper.DelayMicroseconds(30, true);
+            DelayHelper.DelayMicroseconds(40, true);
 
-            _controller.SetPinMode(_pin, PinMode.Input);
+            _controller.SetPinMode(_pin, PinMode.InputPullUp);
 
-            // DHT corresponding signal - LOW - 80 microseconds
+            // DHT corresponding signal - LOW - about 80 microseconds
+            count = _loopCount;
```

Comment:
any chance you could use similar logic DelayHelper is doing here? I'd prefer to avoid adding magic  loops since they will likely not work correctly if we add new board. If it doesn't work please at least add a comment what board this was measured on and if you use Debug or Release build

Reply:
also if you're leaving as is also add a comment in the README file that I2C was tested only on `<board name here>` and might not work elsewhere and perhaps open an issue tracking that

Reply:
This loop is considered from the timing of DHT, just to prevent deadly loops caused by time chaos. `_loopCount` does not need to be precise. And it can't wait directly with DelayHelper, because this 80 microseconds is also about, not exactly accurate.

Reply:
Sensor timing can be affected by many factors, such as wire, voltage, temperature, humidity and so on. So actually I don't like using 1-wire protocol very much. Now most of the new sensors produced by the company are I2C protocols.😄